diff -rup src.orig/mudlet-lua/lua/LuaGlobal.lua src/mudlet-lua/lua/LuaGlobal.lua
--- src.orig/mudlet-lua/lua/LuaGlobal.lua	2013-05-14 14:42:53.000000000 +0200
+++ src/mudlet-lua/lua/LuaGlobal.lua	2013-05-14 15:01:21.000000000 +0200
@@ -108,7 +108,7 @@ local packages = {
 	}

 for _, package in ipairs(packages) do
-	local result, msg = pcall(dofile, "./mudlet-lua/lua/" .. package)
+	local result, msg = pcall(dofile, "/usr/share/mudlet/lua/" .. package)
 	if not result then echo("Error attempting to load file: " .. package .. ": "..msg.."\n") end
 end

diff -rup src.orig/TLuaInterpreter.cpp src/TLuaInterpreter.cpp
--- src.orig/TLuaInterpreter.cpp	2013-05-14 14:42:52.000000000 +0200
+++ src/TLuaInterpreter.cpp	2013-05-14 14:45:52.000000000 +0200
@@ -10295,7 +10295,7 @@ void TLuaInterpreter::loadGlobal()
 {
     //QString path = QDir::homePath()+"/.config/mudlet/mudlet-lua/lua/LuaGlobal.lua";

-    QString path = "mudlet-lua/lua/LuaGlobal.lua";
+    QString path = "/usr/share/mudlet/lua/LuaGlobal.lua";
     int error = luaL_dofile( pGlobalLua, path.toLatin1().data() );
     if( error != 0 )
     {
