diff -rup mudlet-src-orig/src/src.pro mudlet-src-patched/src/src.pro
--- mudlet-src-orig/src/src.pro 2016-11-13 21:10:33.000000000 +0100
+++ mudlet-src-patched/src/src.pro      2017-02-28 13:48:52.845553010 +0100
@@ -98,14 +98,12 @@ unix:!macx {
 # Again according to FHS /usr/local/share/games is the corresponding place for locally built games documentation:
     isEmpty( DOCDIR ) DOCDIR = $${DATAROOTDIR}/doc/mudlet
     LIBS += -lpcre \
-        -llua5.1 \
+        -llua \
         -lhunspell \
-        -L/usr/local/lib/ \
         -lyajl \
         -lGLU \
         -lzip \
         -lz
-    INCLUDEPATH += /usr/include/lua5.1
     LUA_DEFAULT_DIR = $${DATADIR}/lua
 } else:win32: {
     LIBS += -L"C:\\mudlet5_package" \
