--- src/mpDris2.in.orig	2014-07-03 09:23:31.033271465 +0200
+++ src/mpDris2.in	2014-07-03 09:34:04.810784693 +0200
@@ -617,7 +617,10 @@
     def playlists(self):
         if not self._playlists:
             names = map(lambda item: item['playlist'], self._perform_mpd_command('listplaylists'))
-            self._playlists = dict(("/%s/%s" % (re.sub(" ", "_", name), re.sub("-", "_", str(name.__hash__()))), name) for name in names)
+            self._playlists = dict(("/%s/%s" % (
+                        re.sub("[^a-zA-Z0-9_]", "_", name),
+                        re.sub("-", "_", str(name.__hash__()))
+                    ), name) for name in names)
         return self._playlists

     def switchtoplaylist(self, playlist_id):
